scripts/inject-deterministic.py

```diff
@@ -4,14 +4,129 @@
 import time
 from mitmproxy import ctx
 
+import hashlib
+
 
 class AddDeterministic():
 
+    def get_script_with_nonche(self, script, nonce=""):
+        """
+         Given a nonce, getScriptWithNonce returns the injected script text with the nonce.
+        If nonce is an empty string, getScriptWithNonce returns the script block
+        without attaching a nonce attribute.
+        Some responses may specify a nonce inside their Content-Security-Policy,
+        script-src directive.
+        The script injector needs to set the injected script's nonce attribute to
+        open execute permission for the injected script.
+        """
+
+        script_with_nouce = "<script"
+
+        if not nonce == "":
+            script_with_nouce += ' nonce="%s"' % nonce
+
+        script_with_nouce += ">"
+        script_with_nouce += script
+        script_with_nouce += "</script>"
+
+        return script_with_nouce
```

Comment:
Let's have `nonce` default to `None` as it's more pythonic. I believe we can also simplify this to:

```python
if nonce:
    return '<script nonce="{}">{}</script>'.format(nonce, script)
return '<script>{}</script>'.format(script)
```